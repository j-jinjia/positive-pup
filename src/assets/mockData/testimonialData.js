const testimonials = [
  {
    name: "Gemma Powell & Max",
    date: "JAN 2020",
    course: "LITTLE STARS COURSE",
    comment:
      "We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on our toes!!! Thank you so much Mel you are the best!!!",
  },
  {
    name: "Gemma Powell & Max1",
    date: "JAN 2020",
    course: "LITTLE STARS COURSE",
    comment:
      "We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on our toes!!! Thank you so much Mel you are the best!!!",
  },
  {
    name: "Gemma Powell & Max2",
    date: "JAN 2020",
    course: "LITTLE STARS COURSE",
    comment:
      "We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on our toes!!! Thank you so much Mel you are the best!!!",
  },
  {
    name: "Gemma Powell & Max3",
    date: "JAN 2020",
    course: "LITTLE STARS COURSE",
    comment:
      "We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on our toes!!! Thank you so much Mel you are the best!!!",
  },
  {
    name: "Gemma Powell & Max4",
    date: "JAN 2020",
    course: "LITTLE STARS COURSE",
    comment:
      "We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on our toes!!! Thank you so much Mel you are the best!!!",
  },
  {
    name: "Gemma Powell & Max5",
    date: "JAN 2020",
    course: "LITTLE STARS COURSE",
    comment:
      "We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on our toes!!! Thank you so much Mel you are the best!!!",
  },
  {
    name: "Gemma Powell & Max6",
    date: "JAN 2020",
    course: "LITTLE STARS COURSE",
    comment:
      "We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on our toes!!! Thank you so much Mel you are the best!!!",
  },
];

export default testimonials;
