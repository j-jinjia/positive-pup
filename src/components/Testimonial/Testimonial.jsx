import "./Testimonial.scss";
import calendar from "../../assets/svgs/calendar.svg";
import folder from "../../assets/svgs/testimonial_course.svg";
import quotations from "../../assets/svgs/quotations-1.svg";
import { useState } from "react";

const Testimonial = ({ name, date, course, comment, active }) => {
  const cardClass = active ? "testimonial testimonial--active" : "testimonial";
  const [filterComment, setFilterComment] = useState(true);

  const charLimit = 400;
  const filteredComment = () => {
    return comment.slice(0, charLimit) + "...";
  };
  const handleClick = () => {
    setFilterComment(!filterComment);
  };

  return (
    <div className={cardClass} data-testid={name}>
      <h3 className="testimonial__header">{name}</h3>
      <div className="testimonial__details">
        <p className="testimonial__detail">
          <img src={calendar} alt="calendar" /> {date}
        </p>
        <p className="testimonial__detail">
          <img src={folder} alt="" /> {course}
        </p>
      </div>
      <p className="testimonial__comment">
        {comment.length > charLimit && filterComment
          ? filteredComment()
          : comment}
        {comment.length > charLimit && (
          <button className="testimonial__comment-button" onClick={handleClick}>
            {filterComment ? "show more" : "show less"}
          </button>
        )}
      </p>
      <img
        className="testimonial__quotations testimonial__quotations--1"
        src={quotations}
        alt="quotation marks"
      />
      <img
        className="testimonial__quotations testimonial__quotations--2"
        src={quotations}
        alt="quotation marks"
      />
    </div>
  );
};

export default Testimonial;
