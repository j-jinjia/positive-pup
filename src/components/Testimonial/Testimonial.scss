@use "../../styles/variables" as *;

.testimonial {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-shrink: 0;
  background: #ffffff;
  box-shadow: 0px 20px 40px rgba(55, 175, 162, 0.14);
  border-radius: 16px;
  padding: 2rem 2.5rem;
  max-width: 34rem;

  &__header {
    font-size: $font-size-h3;
    color: $colour-primary;
  }

  &__details {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 20px;
    color: $colour-quatenary;
  }

  &__detail {
    font-size: 10px;
  }

  &__comment {
    line-height: 150%;
    text-align: center;
    margin-top: 30px;
    color: $colour-quinary;
  }

  &__quotations {
    z-index: 1;
    &--1 {
      position: absolute;
      top: 0;
      left: 0;
      transform: translate(-50%, -50%);
    }

    &--2 {
      position: absolute;
      bottom: 0;
      right: 0;
      transform: translate(50%, 50%) scale(-1, -1);
    }
  }
}

@media screen and (max-width: $tablet-breakpoint) {
  .testimonial {
    display: none;
  }
}
