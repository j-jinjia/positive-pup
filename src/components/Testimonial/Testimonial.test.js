import Testimonial from "../../components/Testimonial/Testimonial";
import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import testimonialData from "../../assets/mockData/testimonialData";

it("should match inactive Testimonial component snapshot", () => {
  const testimonial = testimonialData[0];
  const { container } = customRender(
    <Testimonial
      name={testimonial.name}
      date={testimonial.date}
      course={testimonial.course}
      comment={testimonial.comment}
      active={false}
    />
  );

  expect(container).toMatchSnapshot();
});

it("should match active Testimonial component snapshot", () => {
  const testimonial = testimonialData[0];
  const { container } = customRender(
    <Testimonial
      key={testimonial.name}
      name={testimonial.name}
      date={testimonial.date}
      course={testimonial.course}
      comment={testimonial.comment}
      active={true}
    />
  );

  expect(container).toMatchSnapshot();
});

it("if the text has more than the character limit a show more button should be visible", () => {
  const testimonial = testimonialData[0];
  customRender(
    <Testimonial
      key={testimonial.name}
      name={testimonial.name}
      date={testimonial.date}
      course={testimonial.course}
      comment={testimonial.comment}
      active={true}
    />
  );
  const showMoreButton = screen.queryByRole("button");
  expect(showMoreButton).toBeInTheDocument();
});

it("if the text has less than the character limit a show more button should not be visible", () => {
  const testimonial = testimonialData[2];
  customRender(
    <Testimonial
      key={testimonial.name}
      name={testimonial.name}
      date={testimonial.date}
      course={testimonial.course}
      comment={testimonial.comment}
      active={true}
    />
  );
  const showMoreButton = screen.queryByRole("button");
  expect(showMoreButton).toBeNull();
});

it("if show more button is clicked, the button should update", () => {
  const testimonial = testimonialData[0];
  customRender(
    <Testimonial
      key={testimonial.name}
      name={testimonial.name}
      date={testimonial.date}
      course={testimonial.course}
      comment={testimonial.comment}
      active={true}
    />
  );
  const showToggle = screen.getByRole("button");
  userEvent.click(showToggle);
  expect(showToggle).toHaveTextContent("show less");
  userEvent.click(showToggle);
  expect(showToggle).toHaveTextContent("show more");
});
