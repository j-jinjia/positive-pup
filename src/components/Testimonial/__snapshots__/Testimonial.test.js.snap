// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match active Testimonial component snapshot 1`] = `
<div>
  <div
    class="testimonial testimonial--active"
    data-testid="Gemma Powell & Max"
  >
    <h3
      class="testimonial__header"
    >
      Gemma Powell & Max
    </h3>
    <div
      class="testimonial__details"
    >
      <p
        class="testimonial__detail"
      >
        <img
          alt="calendar"
          src="calendar.svg"
        />
         
        JAN 2020
      </p>
      <p
        class="testimonial__detail"
      >
        <img
          alt=""
          src="testimonial_course.svg"
        />
         
        LITTLE STARS COURSE
      </p>
    </div>
    <p
      class="testimonial__comment"
    >
      We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on...
      <button
        class="testimonial__comment-button"
      >
        show more
      </button>
    </p>
    <img
      alt="quotation marks"
      class="testimonial__quotations testimonial__quotations--1"
      src="quotations-1.svg"
    />
    <img
      alt="quotation marks"
      class="testimonial__quotations testimonial__quotations--2"
      src="quotations-1.svg"
    />
  </div>
</div>
`;

exports[`should match inactive Testimonial component snapshot 1`] = `
<div>
  <div
    class="testimonial"
    data-testid="Gemma Powell & Max"
  >
    <h3
      class="testimonial__header"
    >
      Gemma Powell & Max
    </h3>
    <div
      class="testimonial__details"
    >
      <p
        class="testimonial__detail"
      >
        <img
          alt="calendar"
          src="calendar.svg"
        />
         
        JAN 2020
      </p>
      <p
        class="testimonial__detail"
      >
        <img
          alt=""
          src="testimonial_course.svg"
        />
         
        LITTLE STARS COURSE
      </p>
    </div>
    <p
      class="testimonial__comment"
    >
      We would just like to say how AMAZING Mel is at The Positive Pup! We have just finished the Rising Stars course and Mel was brilliant. Our black lab Max was an over excitable dog who needed lots of mental stimulation and clear instructions. We have learnt so much and would urge anyone who is looking for dog training to look no further. We are already looking into more courses to keep us and Max on...
      <button
        class="testimonial__comment-button"
      >
        show more
      </button>
    </p>
    <img
      alt="quotation marks"
      class="testimonial__quotations testimonial__quotations--1"
      src="quotations-1.svg"
    />
    <img
      alt="quotation marks"
      class="testimonial__quotations testimonial__quotations--2"
      src="quotations-1.svg"
    />
  </div>
</div>
`;
